tests/js/plugin/texthighlight-test.coffee

```diff
@@ -0,0 +1,78 @@
+assert = chai.assert
+sinon.assert.expose(assert, prefix: '')
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+describe 'Annotator.Plugin.TextHighlight', ->
+  sandbox = null
+  jqElement = null
+  scrollTarget = null
+
+  createTestHighlight = ->
+    anchor =
+      id: "test anchor"
+      annotation: "test annotation"
+      anchoring:
+        id: "test anchoring manager"
+        annotator:
+          id: "test annotator"
+          element:
+            delegate: sinon.spy()
+
+    new Annotator.TextHighlight anchor, "test page", "test range"
+
+  beforeEach ->
+    sandbox = sinon.sandbox.create()
+    sandbox.stub Annotator.TextHighlight, 'highlightRange',
+      (normedRange, cssClass) -> "test highlight span"
+
+    sandbox.stub Annotator.$.fn, "init", (selector, context) ->
```

Comment:
> I don't understand what's happening here. What is the stubbing of `Annotator.$.fn.init`?

This is how I capture `$(<whatever>)` calls. Like [this one](https://github.com/hypothesis/h/blob/add-highlight-tests/h/static/scripts/annotator/plugin/texthighlights.coffee#L77) or [this one](https://github.com/hypothesis/h/blob/add-highlight-tests/h/static/scripts/annotator/plugin/texthighlights.coffee#L120).

> When is it called to set `jqElement`?

It's not called with the purpose of setting `jqElement`; it's called (by the TextHighlight plugin) to do something jQuery-ish. We are only saving the created `jqElement` as a way to faciliate testing, for example to gain access to the created spies. We use this in the tests [here](https://github.com/hypothesis/h/blob/add-highlight-tests/tests/js/plugin/texthighlight-test.coffee#L57), [here](https://github.com/hypothesis/h/blob/add-highlight-tests/tests/js/plugin/texthighlight-test.coffee#L61) and [here](https://github.com/hypothesis/h/blob/add-highlight-tests/tests/js/plugin/texthighlight-test.coffee#L68).

   * * *

Btw, this is the part I was referring to in the OP when I wrote that

> I had to do some pretty ugly mocking around jQuery, if someone can improve it, please tell me how.